### PR TITLE
[#126252] Could not upload files when adding to existing order

### PR DESCRIPTION
**Pull request by jhanggi**
This was broken by #543

### CODE REVIEW

File: spec/controllers/order_details_controller_spec.rb [hunk 1]
```diff
@@ -346,4 +346,33 @@ def perform_update
...
+      it "has access" do
+        get :order_file, params
+        expect(response).to be_success
+      end
+    end
+
+    describe "adding to an existing order" do
+      let(:order) { create(:purchased_order, product: product) }
+      let(:user) { create(:user, :staff, facility: facility) }
+      let(:merge_order) { create(:setup_order, product: product, account: order.account, user: order.user, merge_with_order_id: order.id) }
```
Comment (drench) [inline spec/controllers/order_details_controller_spec.rb]: More of a `TODO`, but merge order setup is complex enough to become a `trait` of `:setup_order`.

**Comment by drench** [comment]
:+1:, with the `:trait` suggestion if it's worth taking on now.

File: spec/controllers/order_details_controller_spec.rb (hunk 1 above)
Comment (jhanggi) [inline spec/controllers/order_details_controller_spec.rb]: Updated. I went with making `merge_order` a full factory. I feel like `create(:merge_order)` is easier to understand than `create(:setup_order, :merge_order)`

**Comment by drench** [comment]
:+1:

**Pull request merged**